apps/docs/pages/guides/database/large-datasets.mdx

```diff
@@ -0,0 +1,128 @@
+
+You can use it in conjunction with Supabase by following these steps:
+
+1. Install pgloader on your local machine or a server. For more info, you can refer to the official pgloader installation page available at: https://pgloader.readthedocs.io/en/latest/install.html
+
+```jsx
+$ apt-get install pgloader
+```
+
+2. Create a configuration file that specifies the source data and the target Supabase database (e.g., config.load).
+
+Here's an example configuration file:
+
+```sql
+LOAD DATABASE
+    FROM sourcedb://USER:PASSWORD@HOST/SOURCE_DB
+    INTO postgres://postgres:password@db.xxxx.supabase.co:6543/postgres
+ALTER SCHEMA 'public' OWNER TO 'postgres';
+set wal_buffers = '64MB', max_wal_senders = 0, statement_timeout = 0, work_mem to '2GB';
+```
+
+Customize the source and Supabase database URL and options to fit your specific use case:
+
+- `wal_buffers`: This parameter is set to '64MB' to allocate 64 megabytes of memory for write-ahead logging buffers. A larger value can help improve write performance by caching more data in memory before writing it to disk. This can be useful during data import operations to speed up the writing of transaction logs.
+- `max_wal_senders`: It is set to 0, to disable replication connections. This is done during the data import process to prevent replication-related conflicts and issues.
+- `statement_timeout`: The value is set to 0, which means it's disabled, allowing SQL statements to run without a time limit.
+- `work_mem`: It is set to '2GB', allocating 2 GB of memory for query operations. This enhances the performance of complex queries by allowing larger in-memory datasets.
+
+3. Run pgloader with the configuration file.
+
+```jsx
+pgloader config.load
+```
+
+For databases using the Postgres engine, we recommend using the [pg_dump](https://www.postgresql.org/docs/current/app-pgdump.html) and [psql](https://www.postgresql.org/docs/current/app-psql.html) command line tools.
+
+### Option 3: Using Postgres Copy command
+
+Read more about [Bulk data loading.](/docs/guides/database/tables#bulk-data-loading)
+
+### Option 4: Using the Supabase API
+
+The Supabase API allows you to programmatically import data into your tables. You can use various client libraries to interact with the API and perform data import operations. This approach is useful when you need to automate data imports, and it gives you fine-grained control over the process. Refer to our [API guide](/docs/guides/api) for more details.
+
+<Admonition type="note">
+When importing data via the Supabase API, it's advisable to refrain from bulk imports. This helps ensure a smooth data transfer process and prevents any potential disruptions.
+
+Read more about [Rate Limiting, Resource Allocation, & Abuse Prevention.](/docs/guides/platform/going-into-prod#rate-limiting-resource-allocation--abuse-prevention)
+
+</Admonition>
+
+## Before you dive into the import process, it's essential to plan and prepare
```

Comment:
```suggestion
## Preparing to import data

Before importing data, you should have a plan for timing, disk space, and more. Prepare your database in advance. 
```

Better to have short, scannable titles and add any necessary intro as text below